nncf/common/logging/track_progress.py

```diff
@@ -0,0 +1,140 @@
+        transient: bool = False,
+        get_time: Optional[Callable[[], float]] = None,
+        refresh_per_second: float = 10,
+        style: StyleType = "bar.back",
+        complete_style: StyleType = "bar.complete",
+        finished_style: StyleType = "bar.finished",
+        pulse_style: StyleType = "bar.pulse",
+        update_period: float = 0.1,
+        disable: bool = False,
+        show_speed: bool = True,
+    ):
+        """
+        Track progress by iterating over a sequence.
+
+        This function is very similar to rich.progress.track(), but with some customizations.
+
+        :param sequence: An iterable (must support "len") you wish to iterate over.
+        :param description: Description of the task to show next to the progress bar. Defaults to "Working".
+        :param total: Total number of steps. Default is len(sequence).
+        :param auto_refresh: Automatic refresh. Disable to force a refresh after each iteration. Default is True.
+        :param transient: Clear the progress on exit. Defaults to False.
+        :param get_time: A callable that gets the current time, or None to use Console.get_time. Defaults to None.
+        :param console: Console to write to. Default creates an internal Console instance.
+        :param refresh_per_second: Number of times per second to refresh the progress information. Defaults to 10.
+        :param style: Style for the bar background. Defaults to "bar.back".
+        :param complete_style: Style for the completed bar. Defaults to "bar.complete".
+        :param finished_style: Style for a finished bar. Defaults to "bar.finished".
+        :param pulse_style: Style for pulsing bars. Defaults to "bar.pulse".
+        :param update_period: Minimum time (in seconds) between calls to update(). Defaults to 0.1.
+        :param disable: Disable display of progress.
+        :param show_speed: Show speed if the total isn't known. Defaults to True.
+        :return: An iterable of the values in the sequence.
+        """
+
+        self.sequence = sequence
+        self.total = total
+        self.description = description
+        self.update_period = update_period
+        self.task = None
+
+        self.columns: List[ProgressColumn] = (
+            [TextColumn("[progress.description]{task.description}")] if description else []
+        )
+        self.columns.extend(
+            (
+                BarColumn(
+                    style=style,
+                    complete_style=complete_style,
+                    finished_style=finished_style,
+                    pulse_style=pulse_style,
+                ),
+                TaskProgressColumn(show_speed=show_speed),
+                IterationsColumn(),
+                SeparatorColumn(),
+                TimeElapsedColumn(),
+                SeparatorColumn(disable_if_no_total=True),  # disable because time remaining will be empty
+                TimeRemainingColumn(elapsed_when_finished=True),
```

Comment:
CI looks good 👍 
```
23:41:10  Statistics collection ━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 300/300 • 0:00:19 • 0:00:19
23:41:10  INFO:nncf:Compiling and loading torch extension: quantized_functions_cpu...
23:41:25  INFO:nncf:Finished loading torch extension: quantized_functions_cpu
23:41:25  Applying Fast Bias correction ━━━━━━━━━━━━━━━━━━━ 100% 52/52 • 0:00:00 • 0:00:00
```

But not `elapsed_when_finished=True` is not needed. On finish prints two same collumns with elapsed time.

Reply:
Done